pkg/kratos/service.go

```diff
@@ -117,6 +117,41 @@ func (s *Service) AcceptLoginRequest(ctx context.Context, identityID string, lc
 	return redirectTo, resp.Cookies(), nil
 }
 
+func (s *Service) GetLoginRequest(ctx context.Context, loginChallenge string) (*hClient.OAuth2LoginRequest, []*http.Cookie, error) {
+	ctx, span := s.tracer.Start(ctx, "kratos.Service.GetLoginRequest")
+	defer span.End()
+
+	redirectTo, resp, err := s.hydra.OAuth2Api().
+		GetOAuth2LoginRequest(ctx).
+		LoginChallenge(loginChallenge).
+		Execute()
+
+	if err != nil {
+		return nil, nil, err
+	}
+
+	return redirectTo, resp.Cookies(), nil
+}
+
+func (s *Service) MustReAuthenticate(ctx context.Context, hydraLoginChallenge string, session *kClient.Session) (bool, error) {
```

Comment:
suggestion: I would add tracing here also since this method might actually invoke an external API (though the GetLoginRequest (which is already traced)

Reply:
Yeah, I wasn't sure if that was needed or not because the `GetLoginRequest` function is doing the external call and that one is already traced, I wasn't sure on which function to start the trace.

You think that both functions should start a trace or only `MustReAuthenticate`?